experiment/reporter.py

```diff
@@ -42,12 +42,12 @@ def output_report(web_bucket, in_progress=False):
         generate_report.generate_report([experiment_name],
                                         str(reports_dir),
                                         in_progress=in_progress)
-        filestore_utils.rsync(str(reports_dir),
-                              web_bucket,
-                              gsutil_options=[
-                                  '-h',
-                                  'Cache-Control:public,max-age=0,no-transform'
-                              ])
+        filestore_utils.rsync(
+            str(reports_dir),
+            web_bucket,
+            options={
+                "gsutil": ['-h', 'Cache-Control:public,max-age=0,no-transform']
```

Comment:
I'm not sure what `'-h', 'Cache-Control:public,max-age=0,no-transform'` does, but would it be possible to just set it always, i.e., pass this to gsutil in the gsutil_util implementation of firestore_utils.rsync()?

Reply:
I don't know whether this could be generalized to all `gsutil rsync`. Other `rsync` don't have this `-h Cache-Control:public,max-age=0,no-transform'.

Reply:
I think the best solution here is to have gsutil_options that only get passed to the gsutil version of this function.

Reply:
It adds the header: `Cache-Control:public,max-age=0,no-transform` to the HTTP response, which does some useful stuff for caching. I'd rather not do it on every rsync. Since this is an option that will only get passed to gsutil I think it's fine to have as `gsutil_options` that get ignored if not using gsutil.

Reply:
I agree with @jonathanmetzman for preserving this only and not generalizing it.
But I prefer to remove `gsutil_options` for `filestore_utils.py`.
I prefer to change `options` as a dictionary.

For example, one `rsync` may have different options for different file storages. Then a dictionary will let it work better when we write the code. Like, for `gsutil`, provide this, for `linux`, provide that, etc.

Reply:
sgtm, thanks!